src/transformers/models/llava/configuration_llava.py

```diff
@@ -88,25 +88,36 @@ def __init__(
         projector_hidden_act="gelu",
         vision_feature_select_strategy="default",
         vision_feature_layer=-2,
-        vocab_size=32000,
         **kwargs,
     ):
         self.ignore_index = ignore_index
         self.image_token_index = image_token_index
         self.projector_hidden_act = projector_hidden_act
+
+        if "vocab_size" in kwargs and not isinstance(text_config, dict):
+            warnings.warn(
+                "The `vocab_size` argument is deprecated and will be removed in v4.42, since it can be inferred from the `text_config`.",
+                FutureWarning,
+            )
+            # set the vocab_size
+            text_config["vocab_size"] = kwargs["vocab_size"]
```

Comment:
No - this is still wrong and will break backwards compatibility

Reply:
vocab size from the kwargs was only ever set when text_config was `None`

Reply:
Just pushed an update which is resolves this. If not, let me know how we can make it backwards compatible.